### PR TITLE
GPU CI deselected tests updates

### PR DESCRIPTION
**Pull request by ethanglaser**
# Description
Disabling certain GPU CI tests and fix for kNN

Changes proposed in this pull request:
-Additions to deselected tests yaml for tests with DEVICE NOT FOUND errors, tests that lead to segmentation faults, and tests that fail due to lack of knn kd tree GPU support
-Fix for gpu knn result options

Latest CI: [job](http://intel-ci.intel.com/ee0b9a96-67d3-f1ea-8680-a4bf010d0e2e)

### CODE REVIEW

File: deselected_tests.yaml [hunk 1]
```diff
@@ -633,3 +633,87 @@ gpu:
   - model_selection/tests/test_validation.py::test_validation_curve_cv_splits_consistency
   - model_selection/tests/test_search.py::test_grid_search_cv_results
   - model_selection/tests/test_search.py::test_random_search_cv_results
+
```
Comment (napetrov) [inline deselected_tests.yaml:636]: Now this would disable tests for both CPU and GPU - we need to change this for GPU only

Comment (napetrov) [reply deselected_tests.yaml:636]: thanks, wasn't scrolling that high.

File: deselected_tests.yaml [hunk 2]
```diff
@@ -633,3 +633,87 @@ gpu:
...
+  # Segmentation faults on GPU
+  - ensemble/tests/test_forest.py::test_forest_classifier_oob
+  - ensemble/tests/test_forest.py::test_forest_regressor_oob
+  - tests/test_common.py::test_search_cv
+
+  # KMeans based (unsupported for GPU)
+  - cluster/tests/test_k_means.py
+  - tests/test_common.py::test_estimators[KMeans()
```
Comment (KulikovNikita) [inline deselected_tests.yaml:644]:
```suggestion
  - tests/test_common.py::test_estimators[KMeans()]
```

**Review by KulikovNikita — APPROVED** [review]
Do we have any plans to reduce number of deselected tests?

File: deselected_tests.yaml (hunk 2 above)
Comment (ethanglaser) [reply deselected_tests.yaml:644]: The way it is currently will deselect all configurations of pytest args that use kmeans (i.e. tests/test_common.py::test_estimators[KMeans()-check_sample_weights_invariance(kind=ones)], and tests/test_common.py::test_estimators[KMeans()-check_sample_weights_pandas_series], etc.) - around 30 in total. By closing the bracket I don't think it will detect any tests.

File: onedal/neighbors/neighbors.py [hunk 3]
```diff
@@ -582,6 +582,8 @@ def _onedal_predict(self, model, X, params, queue):
             model = self._onedal_model
         else:
             model = self._create_model(backend)
+        if 'responses' not in params['result_option'] and gpu_device:
+            params['result_option'] += '|responses'
```
Comment (samir-nasibli) [inline onedal/neighbors/neighbors.py:586]: Please move kNN fix into separate PRs. I think it requires separate review.
[Generally, this part of code should be refactored with no explicitly getting sycl queue device here. This device support logic with fallbacks to daal4py should be resolved on `gpu_support` or `cpu_support` methods like we have for other algos for consistency.] So it is better to move this discussion into separate PR

Comment (ethanglaser) [reply onedal/neighbors/neighbors.py]: Done - https://github.com/intel/scikit-learn-intelex/pull/1337. I think both are ready for approval/merge.
